### PR TITLE
refactor tests + commands

### PR DESCRIPTION
**Pull request by vilit1**
DPS Refactor

command refactor - move registration and compute-device-key under enrollment-group

test refactor - split up the int and unit tests into enrollment and enrollment-group tests

---
This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/). For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.

Thank you for contributing to the IoT extension!

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

Intent for Production

- [x] It is expected that pull requests made to default or core branches such as `dev` or `main` are of production grade. Corollary to this, any merged contributions to these branches may be deployed in a public release at any given time. By checking this box, you agree and commit to the expected production quality of code.

Basic expectations

- [x] If introducing new functionality or modified behavior, are they backed by unit and/or integration tests?
- [x] In the same context as above are command names and their parameter definitions accurate? Do help docs have sufficient content?
- [x] Have **all** the relevant unit **and** integration tests pass? i.e. `pytest <project root> -vv`. Please provide evidence in the form of a screenshot showing a succesful run of tests locally OR a link to a test pipeline that has been run against the change-set.
- [x] Have linter checks passed using the `.pylintrc` and `.flake8` rules? Look at the CI scripts for example usage.
- [x] Have extraneous print or debug statements, commented out code-blocks or code-statements (if any) been removed from the surface area of changes?
- [x] Have you made an entry in HISTORY.rst which concisely explains your user-facing feature or change?

Azure IoT CLI maintainers reserve the right to enforce any of the outlined expectations.

A PR is considered **ready for review** when all basic expectations have been met (or do not apply).

### CODE REVIEW

**Comment by vilit1** [comment]
https://dev.azure.com/azureiotdevxp/aziotcli/_build/results?buildId=5269&view=results

**Comment by vilit1** [comment]
https://dev.azure.com/azureiotdevxp/aziotcli/_build/results?buildId=5274&view=results

File: azext_iot/_help.py [hunk 1]
```diff
@@ -1340,6 +1340,53 @@
...
+    type: command
+    short-summary: Delete a device registration in an Azure IoT Hub Device Provisioning Service.
+"""
+
+helps[
+    "iot dps enrollment-group compute-device-key"
+] = """
+    type: command
+    short-summary: Generate a derived device SAS key.
+    long-summary: Generate a derived device key for a DPS enrollment group.
```
Comment (digimaun) [inline azext_iot/_help.py:1379]: We can eliminate `long-summary` here and combine the bit of additional context with `short-summary`.

File: azext_iot/tests/dps/enrollment_group/test_iot_dps_enrollment_group_int.py [hunk 2]
```diff
@@ -11,397 +11,23 @@
...
-test_endorsement_key = (
-    "AToAAQALAAMAsgAgg3GXZ0SEs/gakMyNRqXXJP1S124GUgtk8qHaGzMUaaoABgCAAEMAEAgAAAAAAAEAibym9HQP9vxCGF5dVc1Q"
-    "QsAGe021aUGJzNol1/gycBx3jFsTpwmWbISRwnFvflWd0w2Mc44FAAZNaJOAAxwZvG8GvyLlHh6fGKdh+mSBL4iLH2bZ4Ry22cB3"
-    "CJVjXmdGoz9Y/j3/NwLndBxQC+baNvzvyVQZ4/A2YL7vzIIj2ik4y+ve9ir7U0GbNdnxskqK1KFIITVVtkTIYyyFTIR0BySjPrRI"
-    "Dj7r7Mh5uF9HBppGKQCBoVSVV8dI91lNazmSdpGWyqCkO7iM4VvUMv2HT/ym53aYlUrau+Qq87Tu+uQipWYgRdF11KDfcpMHqqzB"
-    "QQ1NpOJVhrsTrhyJzO7KNw=="
-)
-
 
 class TestDPSEnrollments(IoTDPSLiveScenarioTest):
```
Comment (digimaun) [inline azext_iot/tests/dps/enrollment_group/test_iot_dps_enrollment_group_int.py:19]: `TestDPSEnrollmentGroups`?

File: azext_iot/_help.py [hunk 3]
```diff
@@ -1340,6 +1340,52 @@
...
+    type: command
+    short-summary: Get a device registration for an enrollment group in an Azure IoT Hub Device
+        Provisioning Service.
+"""
+
+helps[
+    "iot dps enrollment-group registration delete"
+] = """
+    type: command
+    short-summary: Delete a device registration in an Azure IoT Hub Device Provisioning Service.
```
Comment (digimaun) [inline azext_iot/_help.py:1371]: Missing 'enrollment group' verbiage like the other enrollment-group registration commands.

Comment (vilit1) [reply azext_iot/_help.py]: done

**Pull request merged**